### PR TITLE
Fix draft course errors on course builder

### LINKED ISSUE #1508: Course Builder error
### Reproduction Steps

- Create a new course and save as draft

- Then launch the course builder.

- Create a section and create a lesson.

- Right-click the settings icon for the lesson and inspect the console. 

- Here's the screenshot:

image

- When I refreshed, the errors disappear.

https://www.loom.com/share/685539a7b4ef477eb02ad8c9769b6459

### Expected Behavior

Errors should not appear 

### Actual Behavior

Errors appear

### Error Messages / Logs

+ Include any relevant error messages or log files
```
N/A

```

### System and Environment Information

<details>
<summary>System Report</summary>

```
System Report:
Wordpress
-------------------------------------------

Home Url: ht
Site Url: http
Login Url: 
Version: 5.6
Debug Mode: No
Debug Log: No
Debug Display: Yes
Locale: en_US
Multisite: No
Page For Posts: Not Set
Page On Front: Home (#225) 
Permalink Structure: /%postname%/
Show On Front: page
Wp Cron: Yes


Settings
-------------------------------------------

Version: 4.13.0
Db Version: 4.13.0
Course Catalog: Course Catalog (#11) 
Membership Catalog: Membership Catalog (#12)
Student Dashboard: Dashboard (#14) 
Checkout Page:(#13) []
Course Catalog Per Page: 9
Course Catalog Sorting: menu_order
Membership Catalog Per Page: 9
Membership Catalog Sorting: menu_order
Site Membership: Not Set
Courses Endpoint: my-courses
Edit Endpoint:
Lost Password Endpoint: lost-password
Vouchers Endpoint: redeem-voucher
Autogenerate Username: yes
Password Strength Meter: no
Minimum Password Strength: weak
Terms Required: no
Terms Page: Not Set
Checkout Names: required
Checkout Address: required
Checkout Phone: optional
Checkout Email Confirmation: yes
Open Registration: yes
Registration Names: required
Registration Address: optional
Registration Phone: hidden
Registration Voucher: optional
Registration Email Confirmation: yes
Account Names: required
Account Address: required
Account Phone: optional
Account Email Confirmation: yes
Confirmation Endpoint: confirm-payment
Force Ssl Checkout: no
Country: US
Currency: USD
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Decimals: 2
Trim Zero Decimals: no
Recurring Payments: no
Email From Address: 
Email From Name: 
Email Footer Text:
Email Header Image:
Cert Bg Width: 800
Cert Bg Height: 616
Cert Legacy Compat: no


Constants
-------------------------------------------

LLMS_REMOVE_ALL_DATA: undefined
LLMS_REST_DISABLE: undefined
LLMS_SITE_FEATURE_RECURRING_PAYMENTS: undefined
LLMS_SITE_IS_CLONE: undefined


Gateways
-------------------------------------------

Manual: Disabled
Manual Logging:
Manual Order: 1


Server
-------------------------------------------

Mysql Version: 5.5.5
Php Curl: Yes
Php Default Timezone: UTC
Php Fsockopen: Yes
Php Max Input Vars: 5000
Php Max Upload Size: 1 GB
Php Memory Limit: 4096M
Php Post Max Size: 1024M
Php Soap: Yes
Php Suhosin: No
Php Time Limt: 3000
Php Version: 7.3.25-1+0~20201130.73+debian9~1.gbp042074
Software: Apache/2.4.25 (Debian)
Wp Memory Limit: 4096M


Browser
-------------------------------------------

HTTP USER AGENT: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36


Theme
-------------------------------------------

Name: Twenty Nineteen
Version: 1.5
Themeuri: https://wordpress.org/themes/twentynineteen/
Authoruri: https://wordpress.org/
Template:
Child Theme: No
Llms Support: No


Plugins
-------------------------------------------

Classic Editor: 1.6
LifterLMS: 4.13.0


Integrations
-------------------------------------------

BbPress: No
BuddyPress: No


Template Overrides
-------------------------------------------

```

</details>

This issue has be recreated:
+ [ ] Locally
+ [x] On a staging site
+ [x] On a production website
+ [ ] With only LifterLMS and a default theme
HS-151797

### PR DESCRIPTION
**Pull request by thomasplevy**
## Description

Fixes #1508 

## How has this been tested?

+ Manually

## Types of changes
+ Bug fix (realistically this is a hack because backbone is a mess and I don't understand why this is happening but it's fixed with this pr)

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

File: assets/js/builder/Models/Section.js [hunk 1]
```diff
@@ -114,6 +114,26 @@ define( [ 'Collections/Lessons', 'Models/_Relationships' ], function( Lessons, R
 			return this._get_sibling( 'next', circular );
 		},
 
+		/**
+		 * Retrieve a reference to the parent course of the section
+		 *
+		 * @since 3.16.0
```
Comment (thomasplevy) [inline assets/js/builder/Models/Section.js]:
```suggestion
		 * @since [version]
```

File: assets/js/builder/Models/Lesson.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 /**
  * Lesson Model
  *
- * @since    3.13.0
- * @version  3.27.0
+ * @since  3.13.0
```
Comment (eri-trabiccolo) [inline assets/js/builder/Models/Lesson.js]:
```suggestion
 * @since 3.13.0
```

File: assets/js/builder/Models/Lesson.js [hunk 3]
```diff
@@ -107,12 +107,13 @@ define( [ 'Models/Quiz', 'Models/_Relationships', 'Models/_Utilities', 'Schemas/
 		/**
 		 * Retrieve a reference to the parent course of the lesson
 		 *
-		 * @return   obj
-		 * @since    3.16.0
-		 * @version  3.16.0
+		 * @since 3.16.0
+		 * @since [version] Use Section.get_course() in favor of Section.get_parent().
+		 *
+		 * @return obj
```
Comment (eri-trabiccolo) [inline assets/js/builder/Models/Lesson.js]:
```suggestion
		 * @return {Object}
```

File: assets/js/builder/Models/Section.js [hunk 4]
```diff
@@ -114,6 +114,26 @@ define( [ 'Collections/Lessons', 'Models/_Relationships' ], function( Lessons, R
 			return this._get_sibling( 'next', circular );
 		},
 
+		/**
+		 * Retrieve a reference to the parent course of the section
+		 *
+		 * @since [version]
+		 *
+		 * @return obj
```
Comment (eri-trabiccolo) [inline assets/js/builder/Models/Section.js]:
```suggestion
		 * @return {Object}
```

File: assets/js/builder/Models/Section.js [hunk 5]
```diff
@@ -114,6 +114,26 @@ define( [ 'Collections/Lessons', 'Models/_Relationships' ], function( Lessons, R
...
+		 */
+		get_course: function() {
+
+			var parent = this.get_parent();
+
+			// When working with an unsaved draft course the parent isn't properly set on the creation of a section.
+			if ( ! parent ) {
+				this.set_parent( window.llms_builder.CourseModel );
+			}
```
Comment (eri-trabiccolo) [inline assets/js/builder/Models/Section.js]:
```suggestion
			// When working with an unsaved draft course the parent isn't properly set on the creation of a section.
			if ( ! this.get_parent() ) {
				this.set_parent( window.llms_builder.CourseModel );
			}
```
golf

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]
Doc and golf changes requested.
Works fine.

**Pull request merged**